internal/commands/vaultsecrets/run/run.go

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package run
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/consul-template/child"
+	preview_secret_service "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/client/secret_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/stable/2023-06-13/client/secret_service"
+	"github.com/hashicorp/hcp/internal/commands/vaultsecrets/secrets/appname"
+	"github.com/hashicorp/hcp/internal/pkg/cmd"
+	"github.com/hashicorp/hcp/internal/pkg/flagvalue"
+	"github.com/hashicorp/hcp/internal/pkg/format"
+	"github.com/hashicorp/hcp/internal/pkg/heredoc"
+	"github.com/hashicorp/hcp/internal/pkg/iostreams"
+	"github.com/hashicorp/hcp/internal/pkg/profile"
+)
+
+type RunOpts struct {
+	Ctx     context.Context
+	Profile *profile.Profile
+	IO      iostreams.IOStreams
+	Output  *format.Outputter
+
+	App           string
+	Command       string
+	PreviewClient preview_secret_service.ClientService
+	Client        secret_service.ClientService
+}
+
+func NewCmdRun(ctx *cmd.Context, runF func(*RunOpts) error) *cmd.Command {
+	opts := &RunOpts{
+		Ctx:           ctx.ShutdownCtx,
+		Profile:       ctx.Profile,
+		IO:            ctx.IO,
+		Output:        ctx.Output,
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+		Client:        secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "run",
+		ShortHelp: "Run a process with secrets from a Vault Secrets app.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp vault-secrets run" }} command lets you
+		run the provided command as a child process while injecting
+        all of the app's secrets as environment variables, with all secret names
+        converted to upper-case. The stdout/stderr from the child
+        process are forwarded to the top level 'hcp vault-secrets run' command.
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Inject secrets as an environment variable:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets run --command="env"
+				`),
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "command",
+					DisplayValue: "COMMAND",
+					Description:  "Defines the invocation of the child process to inject secrets to.",
+					Value:        flagvalue.Simple("", &opts.Command),
+					Required:     true,
+				},
+				{
+					Name:         "app",
+					DisplayValue: "APP",
```

Comment:
this should be NAME.

Reply:
This was from @dadgar - we need to support autocomplete for app name here. Kindly, search for AutoComplete under vaultsecrets folder and get an idea on how to use that.